blocks/ai-image/src/app/generate-image/image-preview.tsx

```diff
@@ -212,352 +202,279 @@ export const ImagePreview = ({
+                    setSelectedImageIndex(null);
+                    setAnimationStage(false);
+                  }, 500);
+                }}
+              >
+                <Box mt={2}>
+                  <ImageDetails
+                    generatedAt={selectedImageGeneratedAt}
+                    isMobile={isMobile}
+                    onSubmit={() => {
+                      const selectedImageEntityId =
+                        selectedImageIndex !== null &&
+                        images[selectedImageIndex]?.entityId;
+
+                      if (selectedImageEntityId) {
+                        onConfirm(selectedImageEntityId);
+                      }
+                    }}
+                    onCancel={() => {
+                      setAnimationStage(ANIMATION_STAGES.DETAILS_FADE_OUT);
                     }}
                   />
-                  Return to options
-                </Button>
-              </Box>
-            </Stack>
-          </Fade>
+                </Box>
+              </Collapse>
+            )}
+          </Box>
         </Box>
-      </Stack>
 
-      <Collapse in={!selectedImageEntity}>
-        <Box
-          sx={({ palette }) => ({
-            boxSizing: "border-box",
-            width: 1,
-            display: "flex",
-            alignItems: "center",
-            justifyContent: "space-between",
-            border: `1px solid ${palette.gray[30]}`,
-            background: palette.gray[10],
-            borderBottomLeftRadius: 10,
-            borderBottomRightRadius: 10,
-            borderTopWidth: 0,
-            paddingY: 2.125,
-            paddingX: 3.75,
-          })}
-        >
-          <Box display="flex" gap={1}>
-            <ArrowUpIcon
-              sx={{ fontSize: 16, color: ({ palette }) => palette.gray[40] }}
-            />
-            <Typography
-              sx={{
-                color: ({ palette }) => palette.gray[80],
-                fontSize: 14,
-                lineHeight: "18px",
-                fontWeight: 500,
-              }}
+        <Collapse in={selectedImageIndex === null}>
+          <Box
+            sx={({ palette }) => ({
+              boxSizing: "border-box",
+              width: 1,
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "space-between",
+              background: palette.gray[10],
+              paddingY: 2.125,
+              paddingX: 3.75,
+            })}
+          >
+            <Box display="flex" gap={1}>
+              <ArrowUpIcon
+                sx={{ fontSize: 16, color: ({ palette }) => palette.gray[40] }}
+              />
+              <Typography
+                sx={{
+                  color: ({ palette }) => palette.gray[80],
+                  fontSize: 14,
+                  lineHeight: "18px",
+                  fontWeight: 500,
+                }}
+              >
+                {loading
+                  ? "Uploading images..."
```

Comment:
When the text in the bottom left changes to read "Uploading images..." can we instead change it to read "Generating images..." and swap the up-arrow icon to its left for a small circular loading spinner similar to that used in https://github.com/hashintel/hash/pull/2267 (but a lot smaller)?

Reply:
This is perfect - thank you!